### DISCUSSION TITLE
If `baseCpUrl` is set serve cpresources from `baseCpUrl`

### DISCUSSION THREAD

**Discussion by dominikkrulak** [craftcms/cms#12668] [Ideas] [answered]
I think it'd be better/cleaner to serve control panel resources from `baseCpUrl`.

E.g.:

- before `craft.test/cpresources`
- now `cp.craft.test/cpresources`

Advantages:
- `CORS`
- it doesn't have to go for a resource to another host :smirk:

**Comment by dominikkrulak** [comment, marked as answer]
[resourcebasepath](https://craftcms.com/docs/4.x/config/general.html#resourcebasepath) and [resourcebaseurl](https://craftcms.com/docs/4.x/config/general.html#resourcebaseurl) :grimacing:.

**Reply by brandonkelly** [reply]
Resources should be served based on the requested URL by default, unless `@web` and `@webroot` are being explicitly set to the wrong values from the `aliases` setting in `config/general.php`. (They should be set to the base requested URL and the path up to (but not including) `index.php`, respectively.)

**Reply by dominikkrulak** [reply]
I've set it all and it works great! Now I'm serving control panel requests and `cpresources` from `/project/controlpanel/`.
